tst/SkeletalCartesianClosedOpposite.tst

```diff
@@ -0,0 +1,181 @@
+gap> CocartesianCategoriesTest( finsets, i, a, a, alpha, beta : only_primitive_operations := true );;
+
+gap> CocartesianCategoriesTest( finsets, a, t, i, beta, alpha );;
+gap> CocartesianCategoriesTest( finsets, a, t, i, beta, alpha : only_primitive_operations := true );;
+
+##############################################
+# BraidedCartesianCategoriesTest
+##############################################
+
+gap> a := FinSet( 2 );;
+gap> b := FinSet( 3 );;
+
+gap> BraidedCartesianCategoriesTest( finsets, a, b );;
+gap> BraidedCartesianCategoriesTest( finsets, a, b : only_primitive_operations := true );;
+
+gap> a := FinSet( 3 );;
+gap> b := FinSet( 2 );;
+
+gap> BraidedCartesianCategoriesTest( finsets, a, b );;
+gap> BraidedCartesianCategoriesTest( finsets, a, b : only_primitive_operations := true );;
+
+gap> i := InitialObject( finsets );;
+gap> a := FinSet( 4 );;
+
+gap> BraidedCartesianCategoriesTest( finsets, i, a );;
+gap> BraidedCartesianCategoriesTest( finsets, i, a );;
+
+gap> BraidedCartesianCategoriesTest( finsets, a, i );;
+gap> BraidedCartesianCategoriesTest( finsets, a, i : only_primitive_operations := true );;
+
+gap> t := TerminalObject( finsets );;
+gap> a := FinSet( 3 );;
+
+gap> BraidedCartesianCategoriesTest( finsets, t, a );;
+gap> BraidedCartesianCategoriesTest( finsets, t, a : only_primitive_operations := true );;
+
+gap> BraidedCartesianCategoriesTest( finsets, a, t );;
+gap> BraidedCartesianCategoriesTest( finsets, a, t : only_primitive_operations := true );;
+
+##############################################
+# BraidedCocartesianCategoriesTest
+##############################################
+
+gap> a := FinSet( 2 );;
+gap> b := FinSet( 3 );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, a, b );;
+gap> BraidedCocartesianCategoriesTest( finsets, a, b : only_primitive_operations := true );;
+
+gap> a := FinSet( 3 );;
+gap> b := FinSet( 2 );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, a, b );;
+gap> BraidedCocartesianCategoriesTest( finsets, a, b : only_primitive_operations := true );;
+
+gap> i := InitialObject( finsets );;
+gap> a := FinSet( 4 );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, i, a );;
+gap> BraidedCocartesianCategoriesTest( finsets, i, a : only_primitive_operations := true );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, a, i );;
+gap> BraidedCocartesianCategoriesTest( finsets, a, i : only_primitive_operations := true );;
+
+gap> t := TerminalObject( finsets );;
+gap> a := FinSet( 3 );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, t, a );;
+gap> BraidedCocartesianCategoriesTest( finsets, t, a : only_primitive_operations := true );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, a, t );;
+gap> BraidedCocartesianCategoriesTest( finsets, a, t : only_primitive_operations := true );;
+
+##############################################
+# CartesianClosedCategoriesTest
+##############################################
+
+#@if ValueOption( "no_precompiled_code" ) = fail;
```

Comment:
```suggestion
#@if ValueOption( "no_precompiled_code" ) = fail
```

I'm not sure if this makes a difference, but according to the documentation the string after `#@if` should be an expression and a `;` is not well-defined in that context.

Reply:
```suggestion
#@if ValueOption( "no_precompiled_code" ) <> true
```

Ah, and another thing (sorry for not thinking about this before): To make sure the tests actually run it is a good thing to keep a blocklist of cases where they should not run instead of an allowlist of cases where they should run. In case one specifies a wrong condition (like in the case above or when having a typo, e.g. in `"no_precompiled_code"`) this ensures that the tests run and fail noticeably instead of not running and not failing silently.

Reply:
Thank you, a blocklist is definitely smarter. :) And I actually wanted to ask about that semicolon, but forgot about it..